daemon/cluster/executor/container/controller.go

```diff
@@ -370,6 +403,12 @@ func (r *controller) Shutdown(ctx context.Context) error {
 	}
 
 	if err := r.adapter.shutdown(ctx); err != nil {
+		// The container is already stopped or not even started.
+		// don't need to removeNetworks.
+		if isStoppedContainer(err) {
+			log.G(ctx).Debugf("Shutdown(): container %s already stopped.", r.adapter.container.name())
+			return nil
+		}
```

Comment:
Looks like this was already handled by the line below;

```go
if !(isUnknownContainer(err) || isStoppedContainer(err)) {
```

So this is only adding a debug-log? Do we still need that, or did you only need it while working on your patch?

Also wondering; this looks to be the only place where `adapter.shutdown()` is called; should this logic be in that function?

Reply:
Yes, I think we need debug logs, otherwise we don't know what happened on customer's production system.